sql/hive/src/test/scala/org/apache/spark/sql/hive/HiveShowCreateTableSuite.scala

```diff
@@ -177,17 +178,24 @@ class HiveShowCreateTableSuite extends ShowCreateTableSuite with TestHiveSinglet
   }
 
   test("SPARK-24911: keep quotes for nested fields in hive") {
-    withTable("t1") {
-      val createTable = "CREATE TABLE `t1`(`a` STRUCT<`b`: STRING>)"
-      sql(createTable)
-      val shownDDL = sql(s"SHOW CREATE TABLE t1")
-        .head()
-        .getString(0)
-        .split("\n")
-        .head
-      assert(shownDDL == createTable)
+    val originalCreateHiveTable = TestHive.conf.createHiveTableByDefaultEnabled
+    try {
+      TestHive.conf.setConf(SQLConf.LEGACY_CREATE_HIVE_TABLE_BY_DEFAULT_ENABLED, true)
+      withTable("t1") {
+        val createTable = "CREATE TABLE `t1`(`a` STRUCT<`b`: STRING>)"
+        sql(createTable)
+        val shownDDL = sql(s"SHOW CREATE TABLE t1")
+          .head()
+          .getString(0)
+          .split("\n")
+          .head
+        assert(shownDDL == createTable)
```

Comment:
I still need to check this. This test only passed under legacy mode (match `#createHiveTable` rule). And even using `val createTable = "CREATE TABLE `t1`(`a` STRUCT<`b`: STRING>) USING hive"` under non legacy mode(match `#createTable` rule), it fails due to a single whitespace difference: 

```
org.scalatest.exceptions.TestFailedException: "CREATE TABLE `t1`[ ](`a` STRUCT<`b`: STR..." did not equal "CREATE TABLE `t1`[](`a` STRUCT<`b`: STR..."
```

If this really only pass under legacy mode, I think this should also be a behavior change needs to be mentioned somewhere.

Reply:
This doesn't matter. SHOW CREATE TABLE needs to return a SQL that can run. It's OK to have some extra spaces in the SQL.

Reply:
Ah, that's good.